pilot/info/filespec.py

```diff
@@ -42,31 +44,36 @@ class FileSpec(BaseData):
 
     dataset = ""
     ddmendpoint = ""    ## DDMEndpoint name (input or output depending on FileSpec.type)
+    is_user_defined_ddmendpoint = False	  # if ddmendpoint is not set by user, it's True
 
     ## dispatchDblock =  ""       # moved from Pilot1: is it needed? suggest proper internal name?
     ## dispatchDBlockToken = ""   # moved from Pilot1: is it needed? suggest proper internal name?
 
     ## prodDBlock = ""           # moved from Pilot1: is it needed? suggest proper internal name?
-    ## storage_token = "" # prodDBlockToken = ""      # moved from Pilot1: suggest proper internal name (storage token?)
+    storage_token = ""  # prodDBlockToken = ""      # moved from Pilot1: suggest proper internal name (storage token?)
 
     ## local keys
     type = ''          # type of File: input, output of log
     replicas = None    # list of resolved input replicas
     protocols = None   # list of preferred protocols for requested activity
     surl = ''          # source url
     turl = ''          # transfer url
+    protocol_id = None  # id of the protocol to be used to construct turl
+    storage_id = None  # id of the ddmendpoint
+    path_convention = None  # path convention for eventservice inputs/outputs
     mtime = 0          # file modification time
     status = None      # file transfer status value
     status_code = 0    # file transfer status code
     inputddms = []     # list of DDMEndpoint names which will be considered by default (if set) as allowed storage for input replicas
     workdir = None     # used to declare file-specific work dir (location of given local file when it's used for transfer by copytool)
+    is_tar = False     # whether it's a tar file or not
 
     # specify the type of attributes for proper data validation and casting
-    _keys = {int: ['filesize', 'mtime', 'status_code'],
+    _keys = {int: ['filesize', 'mtime', 'status_code', 'protocol_id', 'storage_id', 'path_convention'],
```

Comment:
storage_id and protocol_id are not the input keys, right? so that no needs to run cast and clean functions for them... 

Moreover I think by using both `storage_id` and `ddmendpoint` you over complicate the logic and each time have to apply translation, the question here = why do we need to complicate?:)

Reply:
in general, can you operate with ddmendpoint name instead of storage_id?

Reply:
storage_id is the input key from panda for esmerge jobs.

Ok. I will reorganize it.